src/api/modules/user.ts

```diff
@@ -0,0 +1,41 @@
+import { API } from "@/api/axios";
+import { OAuthProvider } from "@/types/common";
+
+// 회원가입 여부 및 사용자 정보 조회
+export const oAuthApi = async <T = unknown>(params: {
+  provider: OAuthProvider;
+  redirectUri: string;
+  code: string;
+}) => {
+  const res = await API.post<T>(`/oauth`, params);
+
+  if (res.status !== 200) {
+    throw new Error();
+  }
+  return res.data;
+};
+
+export const signInApi = async <T = unknown>(params: {
+  provider: OAuthProvider;
+  oAuthToken: string;
+}) => {
+  const res = await API.post<T>(`/sign-in`, params);
+
+  if (res.status !== 200) {
+    throw new Error();
+  }
+  return res.data;
+};
+
+export const signUpApi = async <T = unknown>(params: {
+  provider: OAuthProvider;
+  oAuthToken: string;
+  userInfo: object;
```

Comment:
정의해둔 UserInfo 타입 사용하면 좋을 것 같아요!

```suggestion
  userInfo: UserInfo;
```

Reply:
확인 감사합니다ㅎㅎ 수정하겠습니다!